scipy/stats/tests/test_continuous_basic.py

```diff
@@ -358,20 +358,28 @@ def check_pdf(distfn, arg, msg):
 
 def check_pdf_logpdf(distfn, args, msg):
     # compares pdf at several points with the log of the pdf
-    points = np.array([0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8])
+    points = np.array([0, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 1])
     vals = distfn.ppf(points, *args)
-    pdf = distfn.pdf(vals, *args)
-    logpdf = distfn.logpdf(vals, *args)
-    pdf = pdf[pdf != 0]
-    logpdf = logpdf[np.isfinite(logpdf)]
-    msg += " - logpdf-log(pdf) relationship"
-    npt.assert_almost_equal(np.log(pdf), logpdf, decimal=7, err_msg=msg)
+    vals = vals[np.isfinite(vals)]
+    with suppress_warnings() as sup:
+        sup.filter(category=RuntimeWarning, message="divide by zero encountered in true_divide")
+        sup.filter(category=RuntimeWarning, message="divide by zero encountered in log")
+        sup.filter(category=RuntimeWarning, message="divide by zero encountered in power")  # gengamma
```

Comment:
What is the impact of suppressing these warnings? 

Would it be safer to implement a list that excludes the `x=0.0` case from certain distributions? 

That seems to be a common strategy used in other unit tests, e.g. complex derivative unit tests, `expect()` method is unstable for infinite skewness and kurtosis etc. 

I am not sure if the use of exclusion/inclusion lists in unit tests is something we would like to promote or curb in scipy but I'd rather have the warnings go through in all cases except for distributions where we've explicitly decided to suppress.

Reply:
I reverted `check_pdf_logpdf` to its original implementation and added `check_pdf_logpdf_at_endpoints` to check the endpoints. It isn't surprising that the computation runs into issues, as the pdf could easily be 0 (or +infinity) at the endpoints. 
Calling `distfn._support_mask(vals)` probably should have removed the need for the warning suppression, but it didn't catch all the problematic cases.